service/javadsl/client/src/main/scala/com/lightbend/lagom/internal/javadsl/client/CircuitBreakerModule.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.javadsl.client
+
+import javax.inject.{ Inject, Provider, Singleton }
+
+import akka.actor.ActorSystem
+import com.lightbend.lagom.internal.client.{ CircuitBreakerConfig, CircuitBreakerMetricsProviderProvider, CircuitBreakers => CircuitBreakers13 }
+import com.lightbend.lagom.internal.javadsl.CircuitBreakersImpl
+import com.lightbend.lagom.internal.spi.CircuitBreakerMetricsProvider
+import com.lightbend.lagom.javadsl.client.CircuitBreakers
+import play.api.inject.{ Binding, Module }
+import play.api.{ Configuration, Environment }
+
+class CircuitBreakerModule extends Module {
+
+  override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] = {
+    Seq(
+      bind[CircuitBreakers].to[CircuitBreakersImpl],
+      // for backward compatibility we still need to provide it for wiring
+      bind[CircuitBreakers13].toProvider[CircuitBreakersProvider],
+      bind[CircuitBreakerMetricsProvider].toProvider[CircuitBreakerMetricsProviderProvider]
+    )
+  }
+}
+
+@Singleton
+class CircuitBreakersProvider @Inject() (
+  system:               ActorSystem,
+  circuitBreakerConfig: CircuitBreakerConfig,
+  metricsProvider:      CircuitBreakerMetricsProvider
+) extends Provider[CircuitBreakers13] {
+  lazy val get = {
+
+    // only in case some 3rd part lib are still wiring the old one.
+    println( // (no logger available here, so good old println)
```

Comment:
Unfortunately we must continue to provide the deprecated version for Guice. 
Because we don't have a logger available at the point we have no other option then using println. 

The intention of this message is to capture the developer's attention in the rare case in which none of his code is using the CircuitBreakers direct, but some 3rd party libs is being included that do use it. In such a case, everything gets wired up at runtime.

I still need to check how this will behave in a real app. Will we get this ugly println warning event when the old `CircuitBreakers` is not used? How lazy is a Guice Provider? I have no idea, need to test it in real.